### DISCUSSION TITLE
How to override defaults set with `styles` prop on a reusable component?

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#2286] [Q&A] [answered]
We use mantine components wrapped as reusable components and pass props to them. This works really well, but sometimes reusable component default styling needs to be overwritten from the outside for certain cases.

I was wondering what the recommended way would be to overwrite component styling? 
We are using `Global` right now as a workaround, but ideally, we would like to simply define `styles` prop on a `Radio` button and then spread it into the reusable `Radio`.

Here is some pseudo code:

```
<CustomRadio
      styles={(theme) => ({
        radio: {
          borderWidth: 2,
        },
      })}
    />
```

```
<Radio
      styles={(theme) => ({
        radio: {
          ...props.styles.radio,
        },
      })}
    />
```

**Comment by rtivital** [comment, marked as answer]
I suggest adding styles on MantineProvider (https://mantine.dev/theming/mantine-provider/#styles-on-mantineprovider) or supporting styles api via createStyles – https://mantine.dev/guides/custom-components/
